receiver/snmpreceiver/scraper.go

```diff
@@ -0,0 +1,559 @@
+		s.logger.Warn("Failed to collect scalar OID metrics", zap.Error(err))
+	} else if len(metricsMap) != 0 {
+		metricsProcessed = true
+		// Add new scalar metrics to the basic ResourceMetrics with no resource attributes
+		for _, value := range metricsMap {
+			value.MoveTo(metricSlice.AppendEmpty())
+		}
+		// Load the map with the basic ResourceMetrics (that possibly
+		// has scalar OID based metrics already on it)
+		resourceMetricsMap[""] = &resourceMetrics
+	}
+
+	// Load this map with the scalar metric map that was just created for the
+	// basic ResourceMetrics. The info in this map is ultimately already
+	// contained within the resourceMetricsMap, but it is more easily accessable
+	// to pull out a specific existing Metric by resource and metric name here
+	metricsByResourceMap := map[string]map[string]*pmetric.Metric{
+		"": metricsMap,
+	}
+	// Try to scrape column OID based metrics.
+	// resourceMetricsMap is passed in as a place for any created resources/metrics to live
+	// metricsByResourceMap is provided as an easy way to check if a metric is already
+	// associated with an existing resource
+	scalarMetricCnt := len(metricsMap)
+	if err := s.scrapeIndexedMetrics(now, resourceMetricsMap, metricsByResourceMap); err != nil {
+		s.logger.Warn("Failed to collect column OID metrics", zap.Error(err))
+	} else if len(metricsByResourceMap[""]) > scalarMetricCnt || len(metricsByResourceMap) > 1 {
+		metricsProcessed = true
+	}
+
+	// Return error if we failed to scrape any metrics
+	md := pmetric.NewMetrics()
+	if !metricsProcessed {
+		return md, errScrape
+	}
+
+	// Put all of the created ResourceMetrics onto a top level Metrics
+	for _, value := range resourceMetricsMap {
+		value.MoveTo(md.ResourceMetrics().AppendEmpty())
+	}
+
+	return md, nil
+}
+
+// scrapeScalarMetrics retrieves all SNMP data from scalar OIDs and turns the returned scalar data
+// into metrics with optional enum attributes
+func (s *snmpScraper) scrapeScalarMetrics(now pcommon.Timestamp, metricsMap map[string]*pmetric.Metric) (err error) {
+	scalarMetricNamesByOID := map[string]string{}
+	scalarMetricOIDs := []string{}
+
+	// Find all metric scalar OIDs
+	// Also create a map of metric names with OID as key so the metric config will be easy to
+	// matchup later with returned data
+	for name, metricCfg := range s.cfg.Metrics {
+		if len(metricCfg.ScalarOIDs) > 0 {
+			for i, oid := range metricCfg.ScalarOIDs {
+				// Data is returned by the client with '.' prefix on the OIDs.
+				// Making sure the prefix exists here in the configs so we can match it up with returned data later
+				if !strings.HasPrefix(oid.OID, ".") {
+					oid.OID = "." + oid.OID
+					s.cfg.Metrics[name].ScalarOIDs[i].OID = oid.OID
+				}
+				scalarMetricOIDs = append(scalarMetricOIDs, oid.OID)
+				scalarMetricNamesByOID[oid.OID] = name
+			}
+		}
+	}
+
+	// If no scalar metric configs, nothing else to do
+	if len(scalarMetricOIDs) == 0 {
+		return nil
+	}
```

Comment:
Is it possible to exit this function early if `len(metricCfg.ScalarOIDs) == 0`? I might be missing the point of this function otherwise.

Reply:
@cpheps not quite. metricCfg.ScalarOIDs is an array on each metric config. so one metric might have this length at 0, and another might be 1, and another might be 2